src/main/scala/sigmastate/utils/Helpers.scala

```diff
@@ -58,7 +58,12 @@ object Helpers {
     * Helper to construct a byte array from a bunch of bytes. The inputs are actually ints so that I
     * can use hex notation and not get stupid errors about precision.
   */
-  def bytesFromInts(bytesAsInts: Int*): Array[Byte] = bytesAsInts.map(_.toByte).toArray
+  def bytesFromInts(bytesAsInts: Int*): Array[Byte] =
+    bytesAsInts.map{ i =>
+      // values from unsigned byte range will be encoded as negative values which is expected here
+      assert(i >= Byte.MinValue && i <= 0xFF, s"$i is out of the signed/unsigned Byte range")
+      i.toByte
+    }.toArray
```

Comment:
@greenhat I'm suspect here we have boxing of all the bytes in a generic sequence. 
Please check this and avoid by using imperative loop. 
In any case this should be imperative loop, because `map` uses CanBuildFrom with growing intermediate buffers, which are not necessary here because we know the resuting length.